client/src/client.rs

```diff
@@ -366,44 +364,38 @@ pub trait RpcApi: Sized {
         self.call("signrawtransaction", handle_defaults(&mut args, &defaults))
     }
 
-    fn sign_raw_transaction_with_key(
+    fn sign_raw_transaction_with_key<B: AsRef<[u8]>>(
         &self,
-        tx: json::HexBytes,
+        tx: B,
         privkeys: &[&str],
```

Comment:
Might be out of scope, but matches the spirit of the PR: why not also make `privkeys` a slice of `PrivateKey`s?

Reply:
Yeah this was done before `PrivateKey` had serde. I think now it has. There should be a todo on some occurrence of that. Might PR that at some point :)